The option to delete a file will only appear if you are the owner of the sketch

s1050613 opened the pull request:
Pretty small compared to the behemoth of the other one!
I’ve used an online PHP shell (oh yes I use an iPad BTW) to test the PHP and it works, despite the very confusing syntax.
I’ve tested the JS with what the meta tag will look like and it functions correctly for when the meta is true and when it’s false.
Still, there may be typos!

Edit: Can’t find the button to link to the issue; I believe I have to put a keyword in the description (which I believe is this). So:
Fixes #87

s1050613 committed:
Update header.html.php

Figured that this would be the best place to put it. I’ve done tests and it doesn’t give any syntax errors, which is good! The long boolean statement is copy-pasted from views/sketch/sketch-buttons.html.php (well I changed != to ==) so hopefully it should function properly.

s1050613 committed:
Update editor.js

Should be that. Pretty simple compared to the other file.

pingskills commented on resources/assets/js/editor.js:
```diff
@@ -1,3 +1,5 @@
+const clientIsAuthor = document.querySelector(`meta[name='client-is-author']`).content === 'true';
```
I don't think we need this as we already have a variable readOnly that tells us this information. Better not to double up.

pingskills commented on resources/assets/js/editor.js:
```diff
@@ -151,7 +153,7 @@ function addTab(file) {
     let text = document.createTextNode(file.filename);
     span.appendChild(text);
-    if (file.filename !== "main.py") {
+    if (file.filename !== "main.py" && clientIsAuthor) {
```
Look for the readOnly variable in this file. We may need to move the declaration of readOnly but we should use this variable for this test if possible so we don't double up.

pingskills commented on views/layout/header.html.php:
```diff
@@ -18,6 +18,7 @@
   <link rel="mask-icon" href="/images/icons/safari-pinned-tab.svg" color="#5bbad5">
   <meta name="msapplication-TileColor" content="#da532c">
   <meta name="theme-color" content="#ffffff">
+  <meta name="client-is-author" content="<?= $personInfo['loggedIn'] && !empty($sketch['person_id']) && $personInfo['details']['person_id'] == $sketch['person_id']? "true" : "false" ?>">
```
Since we already have this boolean statement in sketch-editor.html.php as data-read-only, and we grab this value in editor.js, we don't need to update this file at all.

The less code we have the less bugs we can have! It's better to reduce the amount of code where possible.

pingskills reviewed (commented):
Thanks for getting on to this so quickly. I've written some thoughts about how we might do this another way. Take a look and see if you can make those changes.

pingskills commented:
Hi, in the file views/sketch/sketch-editor.html.php there is already an attribute which tests if the client is the owner:

<div id="editor" data-sketch-id="<?= $this->esc($sketch['sketch_id']); ?>" data-crsf-token="<?= $personInfo['crsfToken'] ?>" data-read-only="<?= $personInfo['loggedIn'] && $personInfo['details']['person_id'] == $sketch['person_id'] ? 0 : 1 ?>"></div>

If we use this then we don't need to update header.html.php

s1050613 commented:
Ah, yes! Wonder why I didn’t think of it being already in the code.
I’ll fix it up now.

s1050613 committed:
Undo changes to header.html.php

Couldn’t find the option to undo a commit so copy-pasted from the parent repository.

s1050613 committed:
Update editor.js

Should be that!

s1050613 commented:
There; that should be it.

pingskills reviewed (approved):
Looks great!
